src/executor/admin/ShowHostsExecutor.cpp

```diff
@@ -18,64 +18,86 @@ folly::Future<Status> ShowHostsExecutor::execute() {
 }
 
 folly::Future<Status> ShowHostsExecutor::showHosts() {
-    static constexpr char kNoPartition[]        = "No valid partition";
+    static constexpr char kNoPartition[] = "No valid partition";
     static constexpr char kPartitionDelimeter[] = ", ";
+
+    auto *shNode = asNode<ShowHosts>(node());
+    auto makeTranditionalResult = [&](const std::vector<meta::cpp2::HostItem> &hostVec) -> DataSet {
+        DataSet v({"Host",
+                   "Port",
+                   "Status",
+                   "Leader count",
+                   "Leader distribution",
+                   "Partition distribution"});
+        for (const auto &host : hostVec) {
+            nebula::Row r({host.get_hostAddr().host,
+                           host.get_hostAddr().port,
+                           meta::cpp2::_HostStatus_VALUES_TO_NAMES.at(host.get_status())});
+            int64_t leaderCount = 0;
+            for (const auto &spaceEntry : host.get_leader_parts()) {
+                leaderCount += spaceEntry.second.size();
+            }
+            std::stringstream leaders;
+            std::stringstream parts;
+            std::size_t i = 0;
+            for (const auto &l : host.get_leader_parts()) {
+                leaders << l.first << ":" << l.second.size();
+                if (i < host.get_leader_parts().size() - 1) {
+                    leaders << kPartitionDelimeter;
+                }
+                ++i;
+            }
```

Comment:
could use `folly::join` to join string list

Reply:
I think folly::join is good for a well prepared, static vector.
But this logical is a little bit complex for join to handle. 
WDYT?